src/main/java/com/alkemy/ong/infrastructure/rest/mapper/UserMapper.java

```diff
@@ -0,0 +1,34 @@
+package com.alkemy.ong.infrastructure.rest.mapper;
+
+import com.alkemy.ong.domain.User;
+import com.alkemy.ong.infrastructure.config.spring.security.PasswordEncoder;
+import com.alkemy.ong.infrastructure.rest.request.UserRequest;
+import com.alkemy.ong.infrastructure.rest.response.UserResponse;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class UserMapper {
+
+  @Autowired
+  private PasswordEncoder passwordEncoder;
+
+  public User toDomain(UserRequest user) {
+    if (user == null) {
+      return null;
+    }
+
+    return User.builder().firstName(user.getFirstName()).lastName(user.getLastName())
+        .email(user.getEmail()).password(passwordEncoder.encoder().encode(user.getPassword()))
+        .build();
+  }
+
+  public UserResponse toResponse(User newUser) {
+    if (newUser == null) {
+      return null;
+    }
+
+    return UserResponse.builder().firstName(newUser.getFirstName()).lastName(newUser.getLastName())
+        .email(newUser.getEmail()).build();
```

Comment:
SUGERENCIA: esto no es parte del style de Google y por lo tanto es valido dejarlo de esta forma. Sin embargo, para aumentar la legibilidad del código se puede insertar saltos en cada atributo del builder para que no sea una cadena de puntos (que es donde se pierde la legibilidad).

Recorda que el trabajo de un desarrollador es 90% del tiempo leer código y un 10% escribirlo.

“Indeed, the ratio of time spent reading versus writing is well over 10 to 1. We are constantly reading old code as part of the effort to write new code. …[Therefore,] making it easy to read makes it easier to write.”
― Robert C. Martin